airqo-platform/AirQo-api
# containerisation of device registry

- What does this PR do?
This PR introduces the docker files used for containerisation of the device-registry.

- What JIRA task is related to this PR?
No link available

- How do I test out this PR?
Git clone this repo `cd AirQo-api/src/device-registry`
Create a docker image locally: `docker build -t us.gcr.io/airqo-250220/airqo-device-registry-api .`
Afterwards, run the newly created image: `docker run -d -p 5000:3000 us.gcr.io/airqo-250220/airqo-device-registry-api`
Using your REST client, test out all the endpoints for the` device-registry`, use port 5000 as specified above or change to what you would like to use. More like: `EXTERNAL-IP:5000/BASE-URL/PARTIAL-URL`. External IPs include:

> 34.78.78.202,
> 34.77.87.81,
> 35.195.26.191,
> 34.76.192.23
> http://localhost

- Which end points should are ready for testing?
All the endpoints in the `device registry` sheet of the [API documentation](https://docs.google.com/spreadsheets/d/1-NGTDBtTcIi8Fg2UYSWjcK6geKrZdopz5kxRFsCEMPs/edit#gid=2081265667).

- Deployment **(OPTIONAL)** 
One could also push to the Google Container Registry using `docker push us.gcr.io/airqo-250220/airqo-device-registry-api`
